addons/html_editor/static/src/core/dom_plugin.js

```diff
@@ -147,37 +154,34 @@ export class DomPlugin extends Plugin {
         }
 
         startNode = startNode || this.shared.getEditableSelection().anchorNode;
-        // If the selection anchorNode is the editable itself, the content
-        // should not be unwrapped.
```

Comment:
This comment was lost even though you are still checking the condition. I would argue to put the comment back above the corresponding line in `shouldUnwrap`.